Update nova.conf.erb config template with Liberty

mjura opened the pull request.

mjura commented:
As we discussed with Tom about [upgrade_levels] options, this will be not used for SC5 -> SOC6 migration.

toabctl commented:
@mjura needs rebase

mjura commented:
Rebased

vuntz commented:
The `osapi_v3` section can be dropped, afaict.

vuntz commented:
Btw, this needs a rebase (you can cheat by just re-applying the bit from https://github.com/crowbar/crowbar-openstack/pull/105/files)

vuntz commented:
FWIW, in other config files, we kept the explicit `rpc_backend = rabbit`; I'd keep it here too, I guess (just in case the default changes later on).

vuntz commented:
I have some doubts about the memcached_servers options. It got moved under keystone_authtoken, but that was used only for novncproxy. So I don't think we need to set the value in keystone_authtoken, but unsure how novncproxy gets this info now...

vuntz commented:
```
# Verify HTTPS connections (boolean value)
insecure = <%= @cinder_insecure ? 'True' : 'False' %>
```
We lost the default value that was in a comment.

vuntz commented:
[keystone_authtoken] region_name should be set.

vuntz commented:
keystone_ec2_insecure should be set.

vuntz commented:
Asking out loud: should we add an attribute for `max_concurrent_builds`?

vuntz commented:
The bits in [neutron] should be updated to use auth_plugin, instead of all the deprecated options.

vuntz commented:
Not your fault, but it seems your sample as well as mine don't have a [vnc] section. I suggest to leave it that way to keep the diff with the sample minimal -- just mentioning it in case people wonder about it too.

vuntz commented:
Can we set [guestfs] debug when to the same value as [DEFAULT] debug?

vuntz commented:
@mjura I don't know if we want to enable the MKS new magic for VMware. This might be worth creating a followup card, what do you think?

dirkmueller commented:
needs rebase

mjura commented:
> The osapi_v3 section can be dropped, afaict.

Why do we want to remove API V3? Even if we want to disable this API shouldn't we keep this configuration options in nova.conf?

vuntz commented:
> > The osapi_v3 section can be dropped, afaict.
> 
> Why do we want to remove API V3? Even if we want to disable this API shouldn't we keep this configuration options in nova.conf?

Because the v3 API doesn't exist anymore and the sample doesn't have this section anymore? :-)

mjura commented:
>>> The osapi_v3 section can be dropped, afaict.

>> Why do we want to remove API V3? Even if we want to disable this API shouldn't we keep this configuration options in nova.conf?

>Because the v3 API doesn't exist anymore and the sample doesn't have this section anymore? :-)

Are you sure, because I was checking this with config-reference
http://docs.openstack.org/liberty/config-reference/content/list-of-compute-config-options.html

Anyway, I'm not insist with this, just only from curiosity

vuntz commented:
> Are you sure, because I was checking this with config-reference
> http://docs.openstack.org/liberty/config-reference/content/list-of-compute-config-options.html

I'm 99% sure. You can grep for osapi_v3 in the code if you want :-)

mjura commented:
Things which were done:
- explicity enabled rpc_backend = rabbit
- adding missing default insecure 
- region_name in [keystone_authtoken] was set
- keystone_ec2_insecure was set
- debug in [guestfs] section was set
- adding mks support trello card was created
- it was rebased, ofcourse 

Things to clarify:
- osapi_v3 section can be dropped
- novncproxy and memcached_servers option
- add an attribute for max_concurrent_builds
- [neutron] should be updated to use auth_plugin

toabctl commented:
```bindir``` is set to ```/home/root/openstack/nova/.tox/genconfig/local/bin```. looks wrong

toabctl commented:
```
 #scheduler_default_filters = RetryFilter,AvailabilityZoneFilter,RamFilter,DiskFilter,ComputeFilter,ComputeCapabilitiesFilter,ImagePropertiesFilter,ServerGroupAntiAffinityFilter,ServerGroupAffinityFilter
+<% if @has_itxt %>scheduler_default_filters = RetryFilter,AvailabilityZoneFilter,RamFilter,ComputeFilter,ComputeCapabilitiesFilter,ImagePropertiesFilter,TrustedFilter<% end %>
```

Here, ```ServerGroupAntiAffinityFilter,ServerGroupAffinityFilter``` are missing if ```has_itxt```is set to True

toabctl commented:
```
 #scheduler_available_filters = nova.scheduler.filters.all_filters
+<% if @has_itxt -%>scheduler_available_filters = nova.scheduler.filters.standard_filters<% end %>
```

Where does this ```nova.scheduler.filters.standard_filters``` comes from? I can't find that in the code.

dirkmueller commented:
```ruby
# Enable guestfs debug (boolean value)
#debug = false
debug = <%= node[:nova][:debug] %>
```

since we're not supporting guestfs, we might want to keep it false.

dirkmueller commented:
```ruby


# Region name of this node (string value)
#os_region_name = <None>

```

might want to set that one.

toabctl commented:
```
 #notification_driver =
+notification_driver = nova.openstack.common.notifier.rpc_notifier
+<%= "notification_driver = ceilometer.compute.nova_notifier" if node.roles.include?("ceilometer-agent") %>
```
That seems to be wrong. According to http://docs.openstack.org/developer/ceilometer/install/manual.html#installing-the-notification-agent, the ```notification_driver``` should be set to ```notification_driver=messagingv2```

dirkmueller commented:
@toabctl we don't support messagingv2 yet, so no.

dirkmueller commented:
There are two changes that we might want to undo:

```ruby
-linuxnet_interface_driver=""
+linuxnet_interface_driver =
-vncserver_listen="0.0.0.0"
+vncserver_listen = 0.0.0.0
```

dirkmueller commented:
Also, the explicit setting of auth_strategy to keystone was lost. we might want to reestablish that.

toabctl commented:
```
 #connection = <None>
+<% if @database_connection -%>
+connection=<%= @database_connection %>
+<% end -%>
```
Why is it in a ```if``` block? Without a connection I assume nova is not working...

mjura commented:
> bindir is set to /home/root/openstack/nova/.tox/genconfig/local/bin. looks wrong

Thanks, I've change this defaults

mjura commented:
> Here, ServerGroupAntiAffinityFilter,ServerGroupAffinityFilter are missing if has_itxtis set to True

It is good question, but for Juno we skipped also these two

mjura commented:
>Where does this nova.scheduler.filters.standard_filters comes from? I can't find that in the code.

Again we are coming to the point that we can't even relay on documentation, because it was taken from  http://docs.openstack.org/liberty/config-reference/content/list-of-compute-config-options.html

mjura commented:
> since we're not supporting guestfs, we might want to keep it false

Look on comment https://github.com/crowbar/crowbar-openstack/pull/98#issuecomment-159639255, but @dirkmueller you are right, I removed this once again

mjura commented:
>> # Region name of this node (string value)
>> #os_region_name = <None>

> might want to set that one.

I set up os_region_name for [barbican] section

mjura commented:
> There are two changes that we might want to undo:
```
-linuxnet_interface_driver=""
+linuxnet_interface_driver =
-vncserver_listen="0.0.0.0"
+vncserver_listen = 0.0.0.0
```

Was added again

mjura commented:
> Also, the explicit setting of auth_strategy to keystone was lost. we might want to reestablish that

Explicitly was added `auth_strategy = keystone`

mjura commented:
> Why is it in a if block? Without a connection I assume nova is not working

It has to stay as it is, look into `chef/cookbooks/nova/recipes/config.rb` there is comment
`don't expose database connection to the compute clients`

mjura commented:
@toabctl and @dirkmueller : thank you for comments I addressed all of them, still left :

- novncproxy and memcached_servers option
- add an attribute for max_concurrent_builds
- [neutron] should be updated to use auth_plugin

dirkmueller commented:
@mjura can you create followup cards under the epic for that?

dirkmueller commented:
Also, we've lost 

``` ruby

 [ssl]
-<%= "ssl_ca_file = #{@ssl_ca_file}" if @ssl_cert_required %>
-ssl_cert_file = <%= @ssl_cert_file %>
-ssl_key_file = <%= @ssl_key_file %>
```

mjura commented:
It is there, but probably it was melted with some other commit. I will check it

```
[ssl]

#
# From oslo.service.sslutils
#

# CA certificate file to use to verify connecting clients. (string value)
# Deprecated group/name - [DEFAULT]/ssl_ca_file
#ca_file = <None>
<%= "ca_file = #{@ssl_ca_file}" if @ssl_cert_required %>

# Certificate file to use when starting the server securely. (string value)
# Deprecated group/name - [DEFAULT]/ssl_cert_file
#cert_file = <None>
cert_file = <%= @ssl_cert_file %>

# Private key file to use when starting the server securely. (string value)
# Deprecated group/name - [DEFAULT]/ssl_key_file
#key_file = <None>
key_file = <%= @ssl_key_file %>
```

mjura commented:
My opinion on `max_concurrent_builds` is that we should be fine with default value, 10 seems to be enough.
```
# Maximum number of instance builds to run concurrently (integer value)
#max_concurrent_builds = 10
```

mjura commented:
I've tested novncproxy with memcached_servers option in [keystone_authtoken]. Novncproxy works fine.

mjura commented:
I've moved deprecated options for novncproxy to [vnc] section

mjura commented:
Rebased

dirkmueller commented:
:+1: 

somehow it feels like we want to squash those commits a bit, like e.g. those that only update the comments/adds default sections into one. but its not a requirement for merging this, just making it easier to distinguish signal from noise.

dirkmueller commented:
-1

mjura commented:
PR was rebased, I moved back `xvpvncproxy_port` and `xvpvncproxy_host` to [DEFAULT]

toabctl commented:
+1

aplanas commented:
+1

toabctl commented:
@dirkmueller oh. recognized your "-1" . is this still valid and if so, why?

mjura commented:
It was only rebased

mjura commented:
@toabctl : @dirkmueller -1 was because of  `xvpvncproxy_port` and `xvpvncproxy_host` in [vnc] section, which I already fixed moving this back to [DEFAULT]

dirkmueller commented:
hmm, in the current PR I still see it in the vnc section...

basically this is the diff that I consider suspicious in the overall review:

```patch

 # value)
 #instance_uuid_format = "[instance: %(uuid)s] "
␣
-# Format string for user_identity field of the logging_context_format_string
-# (string value)
-#logging_user_identity_format = %(user)s %(tenant)s %(domain)s %(user_domain)s %(project_domain)s
-
 # Enables or disables fatal status of deprecations. (boolean value)
 #fatal_deprecations = false
␣
@@ -1416,29 +1494,13 @@
 # exception when timeout expired. (integer value)
 #rpc_poll_timeout = 1
␣
-# Configures zmq-messaging to use broker or not. (boolean value)
-#zmq_use_broker = false
-
-# Minimal port number for random ports range. (port value)
-# Minimum value: 1
-# Maximum value: 65535
-#rpc_zmq_min_port = 49152
-
-# Maximal port number for random ports range. (integer value)
-# Minimum value: 1
-# Maximum value: 65536
-#rpc_zmq_max_port = 65536
-
-# Number of retries to find free port number before fail with ZMQBindError.
-# (integer value)
-#rpc_zmq_bind_port_retries = 100
+# Shows whether zmq-messaging uses broker or not. (boolean value)
+#zmq_use_broker = true
␣
 # Host to locate redis. (string value)
 #host = 127.0.0.1
␣
-# Use this port to connect to redis host. (port value)
-# Minimum value: 1
-# Maximum value: 65535
+# Use this port to connect to redis host. (integer value)
 #port = 6379
␣
 # Password for Redis server (optional). (string value)
@@ -1447,21 +1509,17 @@
 # Size of executor thread pool. (integer value)
 # Deprecated group/name - [DEFAULT]/rpc_thread_pool_size
 #executor_thread_pool_size = 64
+executor_thread_pool_size = 256
␣
 # The Drivers(s) to handle sending notifications. Possible values are
 # messaging, messagingv2, routing, log, test, noop (multi valued)
-# Deprecated group/name - [DEFAULT]/notification_driver
-#driver =
-
-# A URL representing the messaging driver to use for notifications. If not set,
-# we fall back to the same configuration used for RPC. (string value)
-# Deprecated group/name - [DEFAULT]/notification_transport_url
-#transport_url = <None>
+#notification_driver =
+notification_driver = nova.openstack.common.notifier.rpc_notifier
+<%= "notification_driver = ceilometer.compute.nova_notifier" if node.roles.include?("ceilometer-agent") %>


# Specify a timeout after which a gracefully shutdown server will exit. Zero
 # value means endless wait. (integer value)
-#graceful_shutdown_timeout = 60
+#graceful_shutdown_timeout = 0

# Number of cinderclient retries on failed http calls (integer value)
 #http_retries = 3
␣
+# Verify HTTPS connections (boolean value)
+#insecure = false
+insecure = <%= @cinder_insecure ? 'True' : 'False' %>
+
@@ -1725,7 +1795,7 @@
 #
␣
 # Indicate whether this resource may be shared with the domain received in the
-# requests "origin" header. (list value)
+# requests "origin" header. (string value)
 #allowed_origin = <None>
␣
 # Indicate that the actual request can include user credentials (boolean value)
#
␣
 # Indicate whether this resource may be shared with the domain received in the
-# requests "origin" header. (list value)
+# requests "origin" header. (string value)
 #allowed_origin = <None>
␣
 # Indicate that the actual request can include user credentials (boolean value)
 # Deprecated group/name - [DEFAULT]/sql_idle_timeout
 # Deprecated group/name - [DATABASE]/sql_idle_timeout
 # Deprecated group/name - [sql]/idle_timeout
-#idle_timeout = 3600
-
-# Minimum number of SQL connections to keep open in a pool. (integer value)
-# Deprecated group/name - [DEFAULT]/sql_min_pool_size
-# Deprecated group/name - [DATABASE]/sql_min_pool_size
-#min_pool_size = 1
-
-# Maximum number of SQL connections to keep open in a pool. (integer value)
-# Deprecated group/name - [DEFAULT]/sql_max_pool_size
-# Deprecated group/name - [DATABASE]/sql_max_pool_size
-#max_pool_size = <None>
-
-# Maximum number of database connection retries during startup. Set to -1 to
-# specify an infinite retry count. (integer value)
-# Deprecated group/name - [DEFAULT]/sql_max_retries
-# Deprecated group/name - [DATABASE]/sql_max_retries
-#max_retries = 10
-
-# Interval between retries of opening a SQL connection. (integer value)
-# Deprecated group/name - [DEFAULT]/sql_retry_interval
-# Deprecated group/name - [DATABASE]/reconnect_interval
-#retry_interval = 10
-
-# If set, use this value for max_overflow with SQLAlchemy. (integer value)
-# Deprecated group/name - [DEFAULT]/sql_max_overflow
-# Deprecated group/name - [DATABASE]/sqlalchemy_max_overflow
-#max_overflow = <None>
-
-# Verbosity of SQL debugging information: 0=None, 100=Everything. (integer
-# value)
-# Deprecated group/name - [DEFAULT]/sql_connection_debug
-#connection_debug = 0
-
-# Add Python stack traces to SQL as comment strings. (boolean value)
-# Deprecated group/name - [DEFAULT]/sql_connection_trace
-#connection_trace = false
-# If set, use this value for pool_timeout with SQLAlchemy. (integer value)
-# Deprecated group/name - [DATABASE]/sqlalchemy_pool_timeout
-#pool_timeout = <None>
-
-# Enable the experimental use of database reconnect on connection lost.
-# (boolean value)
-#use_db_reconnect = false
-
-# Seconds between retries of a database transaction. (integer value)
-#db_retry_interval = 1
-
-# If True, increases the interval between retries of a database operation up to
-# db_max_retry_interval. (boolean value)
-#db_inc_retry_interval = true
-
-# If db_inc_retry_interval is set, the maximum seconds between retries of a
-# database operation. (integer value)
-#db_max_retry_interval = 10
-
-# Maximum retries in case of connection error or deadlock error before error is
-# raised. Set to -1 to specify an infinite retry count. (integer value)
-#db_max_retries = 20
-
-#
-# From oslo.db
-#
-
-# The file name to use with SQLite. (string value)
-# Deprecated group/name - [DEFAULT]/sqlite_db
-#sqlite_db = oslo.sqlite
-
-# If True, SQLite uses synchronous mode. (boolean value)
-# Deprecated group/name - [DEFAULT]/sqlite_synchronous
-#sqlite_synchronous = true
-
-# The back end to use for the database. (string value)
-# Deprecated group/name - [DEFAULT]/db_backend
-#backend = sqlalchemy
-
-# The SQLAlchemy connection string to use to connect to the database. (string
-# value)
-# Deprecated group/name - [DEFAULT]/sql_connection
-# Deprecated group/name - [DATABASE]/sql_connection
-# Deprecated group/name - [sql]/connection
-#connection = <None>
-
-# The SQLAlchemy connection string to use to connect to the slave database.
#connection = <None>
-
-# The SQLAlchemy connection string to use to connect to the slave database.
-# (string value)
-#slave_connection = <None>
-
-# The SQL mode to be used for MySQL sessions. This option, including the
-# default, overrides any server-set SQL mode. To use whatever SQL mode is set
-# by the server configuration, set this to no value. Example: mysql_sql_mode=
-# (string value)
-#mysql_sql_mode = TRADITIONAL
␣
-# Timeout before idle SQL connections are reaped. (integer value)
-# Deprecated group/name - [DEFAULT]/sql_idle_timeout
-# Deprecated group/name - [DATABASE]/sql_idle_timeout
-# Deprecated group/name - [sql]/idle_timeout
@@ -2565,9 +2594,7 @@
 # Host to locate redis. (string value)
 #host = 127.0.0.1
␣
-# Use this port to connect to redis host. (port value)
-# Minimum value: 1
-# Maximum value: 65535
+# Use this port to connect to redis host. (integer value)
 #port = 6379
␣
 # Password for Redis server (optional). (string value)

+[oslo_messaging_qpid]
+
+#
+# From oslo.messaging
+#
+
+# Use durable queues in AMQP. (boolean value)
+# Deprecated group/name - [DEFAULT]/amqp_durable_queues
+# Deprecated group/name - [DEFAULT]/rabbit_durable_queues
+#amqp_durable_queues = false
+
+# Auto-delete queues in AMQP. (boolean value)
+# Deprecated group/name - [DEFAULT]/amqp_auto_delete
+#amqp_auto_delete = false
+
+# Send a single AMQP reply to call message. The current behaviour since oslo-
+# incubator is to send two AMQP replies - first one with the payload, a second
+# one to ensure the other have finish to send the payload. We are going to
+# remove it in the N release, but we must keep backward compatible at the same
+# time. This option provides such compatibility - it defaults to False in
+# Liberty and can be turned on for early adopters with a new installations or
+# for testing. Please note, that this option will be removed in the Mitaka
+# release. (boolean value)
+#send_single_reply = false
+
+# Qpid broker hostname. (string value)
+# Deprecated group/name - [DEFAULT]/qpid_hostname
+#qpid_hostname = localhost
+
+# Qpid broker port. (integer value)
# Deprecated group/name - [DEFAULT]/qpid_port
+#qpid_port = 5672
+
+# Qpid HA cluster host:port pairs. (list value)
+# Deprecated group/name - [DEFAULT]/qpid_hosts
+#qpid_hosts = $qpid_hostname:$qpid_port
+
+# Username for Qpid connection. (string value)
+# Deprecated group/name - [DEFAULT]/qpid_username
+#qpid_username =
+
+# Password for Qpid connection. (string value)
+# Deprecated group/name - [DEFAULT]/qpid_password
+#qpid_password =
+
+# Space separated list of SASL mechanisms to use for auth. (string value)
+# Deprecated group/name - [DEFAULT]/qpid_sasl_mechanisms
+#qpid_sasl_mechanisms =
+
+# Seconds between connection keepalive heartbeats. (integer value)
+# Deprecated group/name - [DEFAULT]/qpid_heartbeat
+#qpid_heartbeat = 60
+
+# Transport to use, either 'tcp' or 'ssl'. (string value)
+# Deprecated group/name - [DEFAULT]/qpid_protocol
+#qpid_protocol = tcp
+
+# Whether to disable the Nagle algorithm. (boolean value)
+# Deprecated group/name - [DEFAULT]/qpid_tcp_nodelay
+#qpid_tcp_nodelay = true
+
+# The number of prefetched messages held by receiver. (integer value)
+# Deprecated group/name - [DEFAULT]/qpid_receiver_capacity
+#qpid_receiver_capacity = 1
+
+# The qpid topology version to use.  Version 1 is what was originally used by
+# impl_qpid.  Version 2 includes some backwards-incompatible changes that allow
+# broker federation to work.  Users should update to version 2 when they are
+# able to take everything down, as it requires a clean break. (integer value)
+# Deprecated group/name - [DEFAULT]/qpid_topology_version
+#qpid_topology_version = 1
+
+
@@ -2918,26 +3033,19 @@
 # Deprecated group/name - [DEFAULT]/kombu_reconnect_delay
 #kombu_reconnect_delay = 1.0
␣
-# How long to wait a missing client beforce abandoning to send it its replies.
-# This value should not be longer than rpc_response_timeout. (integer value)
-# Deprecated group/name - [DEFAULT]/kombu_reconnect_timeout
-#kombu_missing_consumer_retry_timeout = 5
-
-# Determines how the next RabbitMQ node is chosen in case the one we are
-# currently connected to becomes unavailable. Takes effect only if more than
-# one RabbitMQ node is provided in config. (string value)
-# Allowed values: round-robin, shuffle
-#kombu_failover_strategy = round-robin
+# How long to wait before considering a reconnect attempt to have failed. This
+# value should not be longer than rpc_response_timeout. (integer value)
+#kombu_reconnect_timeout = 60
␣
 # The RabbitMQ broker address where a single node is used. (string value)
 # Deprecated group/name - [DEFAULT]/rabbit_host
 #rabbit_host = localhost
+rabbit_host = <%= @rabbit_settings[:address] %>
␣
-# The RabbitMQ broker port where a single node is used. (port value)
-# Minimum value: 1
-# Maximum value: 65535
+# The RabbitMQ broker port where a single node is used. (integer value)
 # Deprecated group/name - [DEFAULT]/rabbit_port
 #rabbit_port = 5672

-# SSL version to use (valid only if SSL enabled). Valid values are TLSv1 and
-# SSLv23. SSLv2, SSLv3, TLSv1_1, and TLSv1_2 may be available on some
-# distributions. (string value)
-#version = <None>
-
-# Sets the list of available ciphers. value should be a string in the OpenSSL
-# cipher list format. (string value)
-#ciphers = <None>
␣
␣
@@ -3348,6 +3474,17 @@
 # Keymap for VNC (string value)
 # Deprecated group/name - [DEFAULT]/vnc_keymap
 #keymap = en-us
+keymap = <%= node[:nova][:vnc_keymap] %>
+
+# Port that the XCP VNC proxy should bind to (integer value)
+# Minimum value: 1
+# Maximum value: 65535
+#xvpvncproxy_port = 6081
+xvpvncproxy_port = <%= @bind_port_xvpvncproxy %>
+
+# Address that the XCP VNC proxy should bind to (string value)
+#xvpvncproxy_host = 0.0.0.0
+xvpvncproxy_host = <%= @bind_host %>
```

dirkmueller commented:
also somehow gating fails right now..

mjura commented:
@dirkmueller I have found difference, I updated nova stable/liberty branch and generated new sample. In last few weeks some changes were done.

Things which were done:
 - pulled changes for stable/liberty branch
 - generated new nova.conf.sample and made diff
 - option logging_user_identity_format was added back
 - options zmq_use_broker, rpc_zmq_min_port, rpc_zmq_max_port, rpc_zmq_bind_port_retries were updated
 - option notification_driver was removed, I have no idea what option we should use for driver
 - for [database] some options are duplicated, that's way I removed them
 - I removed [oslo_messaging_qpid] section
 - ensured that xvpvncproxy_port and xvpvncproxy_host are removed from [vnc] section

dirkmueller commented:
do not merge before M8

mjura commented:
@dirkmueller : Why? Why? I'm crying

mjura commented:
It was only rebased

toabctl commented:
@mjura there is still a diff for a couple of vnc options (from ```$ diff -u nova.conf.erb-ORG nova.conf.erb|less``` where nova.conf.erb-ORG is the sample config from the package):
```
+instance_usage_audit_period = hour
 
 # Start and use a daemon that can run the commands that need to be run with
 # root privileges. This option is usually enabled on nodes that run nova
@@ -442,40 +456,15 @@
 # Explicitly specify the temporary working directory (string value)
 #tempdir = <None>
 
-# Location of VNC console proxy, in the form
-# "http://127.0.0.1:6080/vnc_auto.html" (string value)
-# Deprecated group/name - [DEFAULT]/novncproxy_base_url
-#novncproxy_base_url = http://127.0.0.1:6080/vnc_auto.html
-
-# Location of nova xvp VNC console proxy, in the form
-# "http://127.0.0.1:6081/console" (string value)
-# Deprecated group/name - [DEFAULT]/xvpvncproxy_base_url
-#xvpvncproxy_base_url = http://127.0.0.1:6081/console
-
-# IP address on which instance vncservers should listen (string value)
-# Deprecated group/name - [DEFAULT]/vncserver_listen
-#vncserver_listen = 127.0.0.1
-
-# The address to which proxy clients (like nova-xvpvncproxy) should connect
-# (string value)
-# Deprecated group/name - [DEFAULT]/vncserver_proxyclient_address
-#vncserver_proxyclient_address = 127.0.0.1
-
-# Enable VNC related features (boolean value)
-# Deprecated group/name - [DEFAULT]/vnc_enabled
-#enabled = true
-
-# Keymap for VNC (string value)
-# Deprecated group/name - [DEFAULT]/vnc_keymap
-#keymap = en-us
-
 # Port that the XCP VNC proxy should bind to (integer value)
 # Minimum value: 1
 # Maximum value: 65535
 #xvpvncproxy_port = 6081

```

toabctl commented:
also ```ssl_cert_file``` and ```ssl_key_file``` are no longer available in the ```[DEFAULT]``` section.:
```
@@ -496,12 +485,6 @@
 # CA certificate file to use to verify connecting clients (string value)
 #ssl_ca_file = <None>
 
-# SSL certificate of API server (string value)
-#ssl_cert_file = <None>
-
-# SSL private key of API server (string value)
-#ssl_key_file = <None>
-
 # Sets the value of TCP_KEEPIDLE in seconds for each server socket. Not
 # supported on OS X. (integer value)
```

I know theses are in the ```[ssl]``` section and the ones from the ssl section should be used but imo there's no reason to remove the unused values from the [DEFAULT] section.

mjura commented:
It was rebased and I fixed @toabctl comments. I've added once again ssl and removed vnc options in `[DEFAULT]` section

mjura commented:
Rebased

dirkmueller commented:
another set of diffs compared to the reference sample: 

```diff
@@ -1466,18 +1548,16 @@
 # Seconds to wait for a response from a call. (integer value)
 #rpc_response_timeout = 60
 
-# A URL representing the messaging driver to use and its full configuration. If
-# not set, we fall back to the rpc_backend option and driver specific
-# configuration. (string value)
-#transport_url = <None>
@@ -1689,10 +1772,15 @@
 
+# Verify HTTPS connections (boolean value)
+#insecure = false
+insecure = <%= @cinder_insecure ? 'True' : 'False' %>

@@ -1853,103 +1954,9 @@
 # If set, use this value for pool_timeout with SQLAlchemy. (integer value)
 # Deprecated group/name - [DATABASE]/sqlalchemy_pool_timeout
 #pool_timeout = <None>
-
-# Enable the experimental use of database reconnect on connection lost.
-# (boolean value)
-#use_db_reconnect = false
-
-# Seconds between retries of a database transaction. (integer value)
-#db_retry_interval = 1
-
-# If True, increases the interval between retries of a database operation up to
-# db_max_retry_interval. (boolean value)
-#db_inc_retry_interval = true
-
-# If db_inc_retry_interval is set, the maximum seconds between retries of a
-# database operation. (integer value)
-#db_max_retry_interval = 10
-
-# Maximum retries in case of connection error or deadlock error before error is
-# raised. Set to -1 to specify an infinite retry count. (integer value)
-#db_max_retries = 20
-
-#
-# From oslo.db
-#
-
-# The file name to use with SQLite. (string value)
-# Deprecated group/name - [DEFAULT]/sqlite_db
-#sqlite_db = oslo.sqlite
-
-# If True, SQLite uses synchronous mode. (boolean value)
-# Deprecated group/name - [DEFAULT]/sqlite_synchronous
-#sqlite_synchronous = true
-
-# The back end to use for the database. (string value)
-# Deprecated group/name - [DEFAULT]/db_backend
-#backend = sqlalchemy
-
-# The SQLAlchemy connection string to use to connect to the database. (string
-# value)
-# Deprecated group/name - [DEFAULT]/sql_connection
-# Deprecated group/name - [DATABASE]/sql_connection
-# Deprecated group/name - [sql]/connection
-#connection = <None>
-
-# The SQLAlchemy connection string to use to connect to the slave database.
-# (string value)
-#slave_connection = <None>
-
-# The SQL mode to be used for MySQL sessions. This option, including the
-# default, overrides any server-set SQL mode. To use whatever SQL mode is set
-# by the server configuration, set this to no value. Example: mysql_sql_mode=
-# (string value)
-#mysql_sql_mode = TRADITIONAL
-
-# Timeout before idle SQL connections are reaped. (integer value)
-# Deprecated group/name - [DEFAULT]/sql_idle_timeout
-# Deprecated group/name - [DATABASE]/sql_idle_timeout
-# Deprecated group/name - [sql]/idle_timeout
-#idle_timeout = 3600
-# Timeout before idle SQL connections are reaped. (integer value)
-# Deprecated group/name - [DEFAULT]/sql_idle_timeout
-# Deprecated group/name - [DATABASE]/sql_idle_timeout
-# Deprecated group/name - [sql]/idle_timeout
-#idle_timeout = 3600
-
-# Minimum number of SQL connections to keep open in a pool. (integer value)
-# Deprecated group/name - [DEFAULT]/sql_min_pool_size
-# Deprecated group/name - [DATABASE]/sql_min_pool_size
-#min_pool_size = 1
-
-# Maximum number of SQL connections to keep open in a pool. (integer value)
-# Deprecated group/name - [DEFAULT]/sql_max_pool_size
-# Deprecated group/name - [DATABASE]/sql_max_pool_size
-#max_pool_size = <None>
-
-# Maximum number of database connection retries during startup. Set to -1 to
-# specify an infinite retry count. (integer value)
-# Deprecated group/name - [DEFAULT]/sql_max_retries
-# Deprecated group/name - [DATABASE]/sql_max_retries
-#max_retries = 10
-
-# Interval between retries of opening a SQL connection. (integer value)
-# Deprecated group/name - [DEFAULT]/sql_retry_interval
-# Deprecated group/name - [DATABASE]/reconnect_interval
-#retry_interval = 10
-
-# If set, use this value for max_overflow with SQLAlchemy. (integer value)
-# Deprecated group/name - [DEFAULT]/sql_max_overflow
-# Deprecated group/name - [DATABASE]/sqlalchemy_max_overflow
-#max_overflow = <None>
-
-# Verbosity of SQL debugging information: 0=None, 100=Everything. (integer
-# value)
-# Deprecated group/name - [DEFAULT]/sql_connection_debug
-#connection_debug = 0
-
-# Add Python stack traces to SQL as comment strings. (boolean value)
-# Deprecated group/name - [DEFAULT]/sql_connection_trace
-#connection_trace = false
-
-# If set, use this value for pool_timeout with SQLAlchemy. (integer value)
-# Deprecated group/name - [DATABASE]/sqlalchemy_pool_timeout
-#pool_timeout = <None>
+<% unless node[:nova][:db][:pool_timeout].nil? -%>
+pool_timeout = <%= node[:nova][:db][:pool_timeout] %>
+<% end -%>
@@ -2550,6 +2595,15 @@
 
 # Use virtio for bridge interfaces with KVM/QEMU (boolean value)
 #use_virtio_for_bridges = true
+<% if @libvirt_type.eql?('kvm') %>use_virtio_for_bridges = true<% end %>
+
+# Use multipath connection of the iSCSI volume (boolean value)
+#iscsi_use_multipath=false
+<%= "iscsi_use_multipath = true" if @libvirt_enable_multipath %>
+
+# Use multipath connection of the iSER volume (boolean value)
+#iser_use_multipath=false
+<%= "iser_use_multipath = true" if @libvirt_enable_multipath %>
```

dirkmueller commented:
Also,it looks like this part got lost:

```ruby
# Driver or drivers to handle sending notifications. (multi
# valued)
#notification_driver=
notification_driver=nova.openstack.common.notifier.rpc_notifier
<%= "notification_driver=ceilometer.compute.nova_notifier" if node.roles.include?("ceilometer-agent") %>
```

dirkmueller commented:
@mjura would it be possible to squash the commits a little? this is getting slightly hairy to review.

mjura commented:
Sync nova config template once again with nova.conf sample from package.

@dirkmueller: `insecure` option in [cells] section is mentioned in http://docs.openstack.org/liberty/config-reference/content/list-of-compute-config-options.html, so then I would like to keep it

@dirkmueller: `iscsi_use_multipath` and `iser_use_multipath` are also part of nova.conf described http://docs.openstack.org/liberty/config-reference/content/list-of-compute-config-options.html, this should stay too

@dirkmueller: `notification_driver=nova.openstack.common.notifier.rpc_notifier` right now should stay `notification_driver = messagingv2` according to http://docs.openstack.org/liberty/install-guide-obs/ceilometer-nova.html

mjura commented:
@dirkmueller: I squashed also commits for you

dirkmueller commented:
@mjura we're not using messagingv2 but messaging so the upstream documentation doesn't apply

dirkmueller commented:
@mjura do you use "@cinder_insecure" then for an insecure option in [cells] ?

Also, the diff for iscsi* related options is:

```diff

-# Number of times to rescan iSCSI target to find volume (integer value)
-#num_iscsi_scan_tries = 5
-
-# Use multipath connection of the iSCSI volume (boolean value)
-#iscsi_use_multipath = false
-
-# The iSCSI transport iface to use to connect to target in case offload support
-# is desired. Default format is of the form <transport_name>.<hwaddress> where
-# <transport_name> is one of (be2iscsi, bnx2i, cxgb3i, cxgb4i, qla4xxx, ocs)
-# and <hwaddress> is the MAC address of the interface and can be generated via
-# the iscsiadm -m iface command. Do not confuse the iscsi_iface parameter to be
-# provided here with the actual transport name. (string value)
-# Deprecated group/name - [DEFAULT]/iscsi_transport
-#iscsi_iface = <None>

```

mjura commented:
Thanks for your comments, I was investigating also problem with `iser_use_multipath` options
https://bugs.launchpad.net/nova/+bug/1532259

I will continue work on this on monday

mjura commented:
@dirkmueller :  I resync once again this config template 
 - `insecure` option moved only in `[cinder]` section
 - fixed options `iscsi_use_multipath` and `iser_use_multipath`
 - I've changed `notification_driver = messaging`

dirkmueller commented:
now we have this in [DEFAULT] section that doesn't belong there:

```patch
+#serialproxy_host = 0.0.0.0
+
+# Port on which to listen for incoming requests (integer value)
+# Minimum value: 1
+# Maximum value: 65535
+#serialproxy_port = 6083
```

dirkmueller commented:
I guess we can live iwth that for now. +1 and merging, since we really need to get moving on the ZVM front and this is blocking me :(

mjura commented:
As I can see following parameters are duplicated
```
# Host on which to listen for incoming requests (string value)
#serialproxy_host = 0.0.0.0

# Port on which to listen for incoming requests (integer value)
# Minimum value: 1
# Maximum value: 65535
#serialproxy_port = 6083

# Host on which to listen for incoming requests (string value)
#html5proxy_host = 0.0.0.0

# Port on which to listen for incoming requests (integer value)
# Minimum value: 1
# Maximum value: 65535
#html5proxy_port = 6082
```

but it seems that it is bug nova.conf generating sample. I will look into it too

mjura commented:
As follow up change I proposed https://github.com/crowbar/crowbar-openstack/pull/192
